crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -622,8 +622,24 @@ define_tedge_config! {
                 #[doku(as = "PathBuf")]
                 #[tedge_config(reader(private))]
                 key_file: Utf8PathBuf,
-            }
-        }
+            },
+        },
+
+        /// The file that will be used as a the server certificate for the File Transfer Service
+        #[tedge_config(example = "/etc/tedge/device-certs/file_transfer_certificate.pem")]
+        #[doku(as = "PathBuf")]
+        cert_path: Utf8PathBuf,
```

Comment:
I will take the time to write scripts and docs on how to setup all these certificates.

I suspect that the current set of settings could be made simpler. For instance, in practice, the `tedge-agent` running on the main device can use the same certificate to connect the c8y auth proxy and to serve files to child devices. Also, on a child device, a user might wrongly set `http.cert_path` instead of `http.client.auth.cert_file`.
It would be good to have a certificate per service instance and to have config settings that reflect that.
Writing scripts and docs will help me to clarify this.

Reply:
Here is first raw version for a script (where all names are hard-coded!):  https://gist.github.com/didier-wenzek/110b618eda6866814dc02fe4f168ff2d

Reply:
> It would be good to have a certificate per service instance and to have config settings that reflect that.

I just hope per-service certificates won't make the setup too complicated for clients. Ideally, a child device should be able to connect to both the `c8y-proxy` and `FTS` using the same client certificate. At least for our docs/demos, having the same root CA sign all the service certificates as well as the client certificates should simplify things I guess.

> Writing scripts and docs will help me to clarify this.

That'd really help. With the agent being a client of the mapper for the proxy and the reverse for the file transfer service, I found the client cert setting twins of both the services really confusing to get my head around and made a lot of errors.

> Here is first raw version for a script (where all names are hard-coded!): https://gist.github.com/didier-wenzek/110b618eda6866814dc02fe4f168ff2d

That really helped and that long list of config settings highlights the complexity as well. It all makes sense, but as a regular user I find it extremely cumbersome.